src/Lecture1.hs

```diff
@@ -94,7 +102,9 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> [a] -> [a]
+subString start end str = take (max 0 (end + 1) - max 0 start) (drop start str)
```

Comment:
@chshersh please review this solution as well :sweat_smile:

Reply:
This solution looks good 👍🏻 
Feel free to merge your PR 🙂 
I believe it's ready 🚢